.github/test/installability-errors-uptodate-test.js

```diff
@@ -0,0 +1,76 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fetch = require('node-fetch');
+
+const InstallableManifestAudit = require('../../lighthouse-core/audits/installable-manifest.js');
+
+describe('installabilityErrors', () => {
+  let chromiumErrorIds;
+
+  beforeAll(async () => {
+    const installableLoggingGitTilesUrl =
+      'https://chromium.googlesource.com/chromium/src/+/master/chrome/browser/installable/installable_logging.cc?format=TEXT';
+    const base64 = await fetch(installableLoggingGitTilesUrl).then(r => r.text());
+
+    let buff = Buffer.from(base64, 'base64');
+    let text = buff.toString('utf-8');
+
+    // Apply some *beautiful* regexes to parse the C++ of https://chromium.googlesource.com/chromium/src/+/master/chrome/browser/installable/installable_logging.cc
+    const handledErrorConsts = [...text.matchAll(/error_id = (k.*?);/g)].map(([_, match]) => match);
+    const errorConstsToIdsCode = [...text.matchAll(/static const char k.*?Id\[\](.|\n)*?;/g)].map(
+      ([match]) => match.replace(/\n/, '')
+    );
+    const errorConstsToIds = errorConstsToIdsCode.map(txt => [
+      txt.match(/char (k.*?)\[/)[1],
+      txt.match(/ "(.*?)"/)[1],
+    ]);
+    const errorConstsToIdsDict = Object.fromEntries(errorConstsToIds);
+
+    chromiumErrorIds = handledErrorConsts.map(varName => {
+      if (!errorConstsToIdsDict[varName]) throw new Error(`Error const (${varName}) not found!`);
+      return errorConstsToIdsDict[varName];
+    });
+  });
+
+  it('should notify us if something changed', () => {
+    expect(chromiumErrorIds.sort()).toMatchInlineSnapshot(`
+      Array [
+        "already-installed",
+        "cannot-download-icon",
+        "ids-do-not-match",
+        "in-incognito",
+        "manifest-display-not-supported",
+        "manifest-display-override-not-supported",
+        "manifest-empty",
+        "manifest-location-changed",
+        "manifest-missing-name-or-short-name",
+        "manifest-missing-suitable-icon",
+        "no-acceptable-icon",
+        "no-icon-available",
+        "no-id-specified",
+        "no-manifest",
+        "no-matching-service-worker",
+        "no-url-for-service-worker",
+        "not-from-secure-origin",
+        "not-offline-capable",
+        "platform-not-supported-on-android",
+        "prefer-related-applications",
+        "prefer-related-applications-only-beta-stable",
+        "start-url-not-valid",
+        "url-not-supported-for-webapk",
+      ]
+    `);
+  });
+
+  it('are each handled explicitly in the gatherer', () => {
+    // expect.arrayContaining is semantically great, but the output sucks
```

Comment:
or...
```suggestion
    const errorStrings = Object.keys(InstallableManifestAudit.UIStrings)
      .filter(key => chromiumErrorIds.includes(key))
      .sort();
    expect(errorStrings).toEqual(chromiumErrorIds);
```

for a nice one-shot output that lists all the missing ones? :)